feature/gribi/otg_tests/vrf_policy_driven_te/vrf_policy_driven_te_test.go

```diff
@@ -0,0 +1,1946 @@
+		InnHdrSrcIP: atePort1.IPv4, InnHdrDstIP: ipv4InnerDst, isInnHdrV4: true})
+
+	flow2 := createFlow(&flowArgs{flowName: "flow2", srcPort: atePort1, dstPortList: dstPorts,
+		outHdrSrcIP: ipv4OuterSrc111, outHdrDstIP: traffiDstIP2, InnHdrSrcIPv6: atePort1.IPv6,
+		InnHdrDstIPv6: ipv6InnerDst, isInnHdrV4: false, outHdrDscp: []uint32{dscpEncapNoMatch}})
+
+	flow3 := createFlow(&flowArgs{flowName: "flow3", srcPort: atePort1, dstPortList: dstPorts,
+		outHdrSrcIP: ipv4OuterSrc111, outHdrDstIP: traffiDstIP3, InnHdrSrcIP: atePort1.IPv4,
+		InnHdrDstIP: ipv4InnerDst, isInnHdrV4: true, outHdrDscp: []uint32{dscpEncapNoMatch}})
+
+	sendTraffic(t, args, portList, []gosnappi.Flow{flow1, flow2, flow3})
+	verifyTraffic(t, args, []string{"flow1", "flow2", "flow3"}, !wantLoss)
+	captureAndValidatePackets(t, args, &packetValidation{portName: portList[0],
+		outDstIP: []string{traffiDstIP1}, inHdrIP: ipv4InnerDst, validateTTL: false, validateDecap: true})
+
+	t.Run("Send traffic to non decap route and verify the behavior", func(t *testing.T) {
+		flow4 := createFlow(&flowArgs{flowName: flowNegTest, srcPort: atePort1, dstPortList: dstPorts,
+			outHdrSrcIP: ipv4OuterSrc111, outHdrDstIP: ipv4OuterDst333, isInnHdrV4: true,
+			InnHdrSrcIP: atePort1.IPv4, InnHdrDstIP: ipv4InnerDst})
+		sendTraffic(t, args, portList, []gosnappi.Flow{flow4})
+		verifyTraffic(t, args, []string{flowNegTest}, !wantLoss)
+		captureAndValidatePackets(t, args, &packetValidation{portName: portList[0],
+			outDstIP: []string{ipv4OuterDst333}, inHdrIP: ipv4InnerDst, validateNoDecap: true})
+	})
+}
+
+// testGribiDecapMatchSrcProtoDSCP is to validate subtest 2.
+// Test-2, match on source, protocol and DSCP, VRF_DECAP hit -> VRF_ENCAP_A miss -> DEFAULT
+func testGribiDecapMatchSrcProtoDSCP(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice, args *testArgs) {
+	t.Helper()
+	cases := []struct {
+		desc           string
+		prefixWithMask string
+	}{{
+		desc:           "Mask Length 24",
+		prefixWithMask: "192.51.100.0/24",
+	}, {
+		desc:           "Mask Length 32",
+		prefixWithMask: "192.51.100.64/32",
+	}, {
+		desc:           "Mask Length 28",
+		prefixWithMask: "192.51.100.64/28",
+	}, {
+		desc:           "Mask Length 22",
+		prefixWithMask: "192.51.100.0/22",
+	}}
+
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			t.Log("Flush existing gRIBI routes before test.")
+			if err := gribi.FlushAll(args.client); err != nil {
+				t.Fatal(err)
+			}
+
+			// Configure GRIBi baseline AFTs.
+			configGribiBaselineAFT(ctx, t, dut, args)
+
+			t.Run("Program gRIBi route", func(t *testing.T) {
+				configureGribiRoute(ctx, t, dut, args, tc.prefixWithMask)
+			})
+			// Send both 6in4 and 4in4 packets. Verify that the packets have their outer
+			// v4 header stripped and are forwarded according to the route in the DEFAULT
+			// VRF that matches the inner IP address.
+			portList := []string{"port8"}
```

Comment:
This could be defined in global var maybe since its used in most tests

Reply:
Please check we are using different port list depending on test.  Its not common across tests.

./vrf_policy_driven_te_test.go:494:     portList := []*ondatra.Port{p1, p2, p3, p4, p5, p6, p7, p8}
./vrf_policy_driven_te_test.go:1476:                    portList := []string{"port8"}
./vrf_policy_driven_te_test.go:1548:                    portList := []string{"port8"}
./vrf_policy_driven_te_test.go:1595:    portList := []string{"port8"}
./vrf_policy_driven_te_test.go:1644:            portList := []string{"port2"}
./vrf_policy_driven_te_test.go:1713:    portList := []string{"port8"}
./vrf_policy_driven_te_test.go:1789:            portList := []string{"port2"}